xla/service/gpu/BUILD

```diff
@@ -919,13 +919,12 @@ alias(
 
 cc_library(
     name = "_nccl_api_impl",
-    srcs = if_cuda_is_configured(
-        ["nccl_api.cc"],
-        ["nccl_api_stub.cc"],
+    srcs = if_gpu_is_configured(
+        ["nccl_api.cc",  "nccl_api_stub.cc"],
```

Comment:
this might be the offending line ... let me try what happens if I only include nccl_api.cc

Reply:
ah, I think I know what's going on... it's the if-else syntax that's supported by `if_cuda_is_configured`, but not by `if_gpu_is_configured`

Reply:
this will require a change to google TSL... but I get 
```An owner of this repository has limited the ability to open a pull request to users that are collaborators on this repository.```

Reply:
In any case, this is what the change would look like
```
commit 55cf004466de5ec97b7dc9f7916c0fe918c3633c (HEAD -> fix_rocm_not_configured, mycopy/fix_rocm_not_configured)
Author: Jens Glaser <glaserj@ornl.gov>
Date:   Mon Jan 22 19:41:05 2024 -0500

    Fix missing alternative in if_rocm_is_configured
    
    this makes it so that if_rocm_is_configured
    and if_cuda_is_configured can be used with the same interface

diff --git a/third_party/gpus/rocm/build_defs.bzl.tpl b/third_party/gpus/rocm/build_defs.bzl.tpl
index 2b4595bb..ee01bf8f 100644
--- a/third_party/gpus/rocm/build_defs.bzl.tpl
+++ b/third_party/gpus/rocm/build_defs.bzl.tpl
@@ -38,7 +38,7 @@ def rocm_version_number():
     """Returns a list of supported GPU architectures."""
     return %{rocm_version_number}
 
-def if_rocm_is_configured(x):
+def if_rocm_is_configured(x, no_rocm = []):
     """Tests if the ROCm was enabled during the configure process.
 
     Unlike if_rocm(), this does not require that we are building with
@@ -46,7 +46,7 @@ def if_rocm_is_configured(x):
     """
     if %{rocm_is_configured}:
       return select({"//conditions:default": x})
-    return select({"//conditions:default": []})
+    return select({"//conditions:default": no_rocm})
 
 def rocm_hipblaslt():
     return %{rocm_is_configured} and %{rocm_hipblaslt}
```

Reply:
I've added the if/else logic as an additional commit... if that fails we can revert to fd54ad3

Reply:
@ddunl for TSL changed

Reply:
not necessary .... looks like the `if_cuda_or_rocm` already exists, with alternative second argument.. just need to replace usage of that function in xla... see latest commit